akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -370,6 +372,7 @@ private[stream] final class VirtualProcessor[T] extends Processor[T, T] {
           try {
             subscriptionStatus.set(s)
             tryOnSubscribe(s, sub)
+            if (isCancelled) throw canNotSubscribeTheSameSubscriberMultipleTimesException
```

Comment:
Yes and no.

The wording I used in #18486 was indeed unfortunate (wrong), as it violates Rule 1.9.
However! The ticket was actually aimed to resolve the case as shown in the snippet of that ticket, and originated from https://groups.google.com/forum/#!searchin/akka-user/https$3A$2F$2Fgithub.com$2Fakka$2Fakka$2Fissues$2F18486%7Csort:relevance/akka-user/EMuOMaXsG84/SgzXdOPmAgAJ

In other words, it is about [Rule 2.12](https://github.com/reactive-streams/reactive-streams-jvm/tree/v1.0.0#2.12) and [Rule 2.13](https://github.com/reactive-streams/reactive-streams-jvm/tree/v1.0.0#2.13) - i.e. onSubscribe can only be called once + "... *for all other situations the only legal way for a Subscriber to signal failure is by cancelling its Subscription. In the case that this rule is violated, any associated Subscription to the Subscriber MUST be considered as cancelled, and the caller MUST raise this error condition in a fashion that is adequate for the runtime environment.*".

The ticket description should be amended to "log error", and this PR should do so then, as Rule 2.13 mandates.
Sounds about right? :)